daemon/cmd/api_limits.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/rate"
+)
+
+const (
+	apiRequestEndpointCreate = "endpoint-create"
+	apiRequestEndpointDelete = "endpoint-delete"
+	apiRequestEndpointGet    = "endpoint-get"
+	apiRequestEndpointPatch  = "endpoint-patch"
+	apiRequestEndpointList   = "endpoint-list"
+)
+
+var apiRateLimitDefaults = map[string]rate.APILimiterParameters{
+	// PUT /endpoint/{id}
+	apiRequestEndpointCreate: {
+		AutoAdjust:                  true,
+		EstimatedProcessingDuration: time.Second * 2,
+		RateLimit:                   0.5,
+		RateBurst:                   4,
+		MaxParallelRequests:         4,
+		SkipInitial:                 5,
+		MaxWaitDuration:             15 * time.Second,
+		Log:                         true,
+	},
+	// DELETE /endpoint/{id}
+	//
+	// No maximum wait time is enforced as delete calls should always
+	// succeed. Permit a large number of parallel requests to minimize
+	// latency of delete calls, on fast systems, an even higher number of
+	// parallel requests will be allowed.  Logging is enabled for
+	// visibility as frequency should be low.
+	apiRequestEndpointDelete: {
+		EstimatedProcessingDuration: 200 * time.Millisecond,
+		AutoAdjust:                  true,
+		MaxParallelRequests:         4,
+		MinParallelRequests:         4,
+		Log:                         true,
+	},
+	// GET /endpoint/{id}/healthz
+	// GET /endpoint/{id}/log
+	// GET /endpoint/{id}/labels
+	// GET /endpoint/{id}/config
+	//
+	// All GET calls to endpoint attributes are grouped together and rate
+	// limited. On a fast system, the limit should become more liberal
```

Comment:
Reworded

Reply:
I don't see the updated rewording.

Reply:
I just noticed, there were two instances of this and I didn't notice that.

I had changed
```
        // DELETE /endpoint/{id}
        //
        // No maximum wait time is enforced as delete calls should always
        // succeed. Permit a large number of parallel requests to minimize
        // latency of delete calls, if the system performance allows for it,
        // the maximum number of parallel requests will grow to a larger number
        // but it will never shrink below 4. Logging is enabled for visibility
        // as frequency should be low.
```

Reply:
Fixed